src/application/services/useNoteSettings.ts

```diff
@@ -146,6 +153,24 @@ export default function (): UseNoteSettingsComposableState {
     });
   };
 
+  /**
+   * Update note cover pisture
+   * @param id - note id
+   * @param data - picture binary data
+   */
+  const updateCover = async (id: NoteId, data: Blob): Promise<void> => {
+    const key = await noteAttachmentService.upload(id, data);
+
+    const { cover } = await noteSettingsService.patchNoteSettingsByNoteId(id, { cover: key });
```

Comment:
it's better to move it to the domain level since it is a business logic

Reply:
done